### PR TITLE
Update Cutter Documentation

### PR DESCRIPTION
**Pull request by Surendrajat**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/developers-docs/first-time.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/developers-docs.html)
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**
This PR is to update Cutter docs. Starting with adding missing dependencies to dev docs and will slowly move towards user docs.

**Test plan (required)**
Build and run docs.

**Closing issues**

### CODE REVIEW

**Comment by karliss** [comment]
Before adding this to the dependency list I would like the need for it to be evaluated.

**Comment by karliss** [comment]
From what I could find curl is only used for downloading capstone when building r2 using .configure/Makefile approach. It can also use wget instead of curl. Meson based r2 build which is also what CUTTER_USE_BUNDLED_RADARE2 use seems to use GIT for getting it. It is also possible to build r2 using system capstone which avoids downloading it completely. 

I propose organizing the dependency list into 3 parts:
* Main dependencies - what we currently have
* Optional but potentially difficult to avoid dependencies (curl and python3-setuptools) 
* Optional depenrencies for extra features (KSyntaxHighlighter, graphviz)

**Comment by Surendrajat** [comment]
KSyntaxHighlighter & graphviz seems like a case that may need separate mention but for me difficult to avoid kinda sounds the same as the main dependencies. Usually, all build instruction pages show required and recommended dependencies together for the convenience (of setting up env with one command).
I also think that the wget option is good, and git one(if it can be replaced in all script) is probably better.

**Comment by karliss** [comment]
You can call them recommended dependencies if you think that it's less confusing.  My reasoning is that they aren't strictly required and removing them doesn't take away from functionality of Cutter. Adding an additional separator especially if there is one already for optional dependencies isn't hard. There are different needs for different people. Most end users probably don't care how and what the the build system does. But people preparing a package for distros or just someone trying to do offline do care at least a little bit and probably don't want build system to curl random source code during build.

File: docs/source/building.rst [hunk 1]
```diff
@@ -25,14 +25,25 @@ On Linux, you will need:
...
+
+On Debian-based Linux distributions, all of these essential packages can be installed with this single command:
 
 ::
 
    sudo apt install git build-essential cmake meson libzip-dev zlib1g-dev qt5-default libqt5svg5-dev qttools5-dev qttools5-dev-tools
 
-On Arch-based Linux distributions, build-essential should be replaced by base-devel:
+.. note::
+ For Ubuntu 18.04 and lower ``meson`` should be installed with ``pip install --upgrade meson``.
```
Comment (karliss) [inline docs/source/building.rst:44]: I wouldn't recommend to `sudo pip install --upgrade`. That can upgrade the meson installed using system package manager which will cause problems. For users not familiar with python `pip install --user` is in my opinion safer. Those that are familiar with Python can use virtualenv if they want but they should be able to figure out themselves.

Comment (Surendrajat) [reply docs/source/building.rst]: Done.

**Comment by Surendrajat** [comment]
@karliss @ITAYC0HEN The `git clone...` instructions were only available under the **Building on Linux** section so I've made a common **Getting the Source** section to avoid repetition.

**Pull request merged**